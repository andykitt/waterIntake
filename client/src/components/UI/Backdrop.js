import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Backdrop = styled.div`
  position: fixed;
  display: flex;
  justify-content: center;
  align-items: center;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background: rgba(0, 0, 0, 0.4);
  z-index: 999;
  overflow-y: scroll;
`;

const backdrop = props => {
  return (
    <Backdrop id="backdrop" onClick={props.onClick}>
      {props.children}
    </Backdrop>
  );
};

backdrop.propTypes = {
  onClick: PropTypes.func.isRequired,
  children: PropTypes.object
};

export default backdrop;
