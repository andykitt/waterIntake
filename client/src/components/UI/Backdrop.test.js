import React from 'react';
import { shallow, mount } from 'enzyme';

import Backdrop from './Backdrop';

const clickFn = jest.fn();

// describe('Backdrop', () => {
//   it('should render correctly', () => {
//     const component = shallow(<Backdrop />);

//     expect(component).toMatchSnapshot();
//   });
// });

describe('Backdrop', () => {
  it('button click should toggle component visibility', () => {
    const component = shallow(<Backdrop onClick={clickFn} />);

    component.find('#backdrop').simulate('click');

    expect(clickFn).toHaveBeenCalled();
  });
});
