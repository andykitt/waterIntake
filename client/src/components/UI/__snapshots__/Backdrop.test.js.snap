// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop /> 1`] = `
<div
  class="sc-bdVaJa hLPwAO"
  data-testid="backdrop"
/>
`;
